app/models/miq_worker/runner.rb

```diff
@@ -489,4 +494,19 @@ def process_title
   def set_process_title
     Process.setproctitle(process_title)
   end
+
+  private
+
+  def skip_heartbeat?
+    ENV["DISABLE_MIQ_WORKER_HEARTBEAT"]
+  end
+
+  def warn_about_heartbeat_skipping
+    puts "**************************************************"
+    puts "WARNING:  SKIPPING HEARTBEATING WITH THIS WORKER!"
+    puts "**************************************************"
+    puts ""
+    puts "Remove the `DISABLE_MIQ_WORKER_HEARTBEAT` ENV variable"
+    puts "to reenable heartbeating normally."
```

Comment:
also, if its a warning, consider using warn instead of puts (https://ruby-doc.org/core-2.4.0/Kernel.html#method-i-warn)